### PR TITLE
[fix] #3548: Fix IntoSchema transparent attribute

### LINKED ISSUE #3548: [bug]: `IntoSchema` doesn't handle `transparent` attribute for non-tuple structs correctly
When we have `schema(transparent)` attached to a type with `IntoSchema` derive it should implement `IntoSchema` as an alias to an underlying type.

However it doesn't work for structs with one field (example: `GenesisTransaction`).

Originall issue was found by @arndey and debugged by my and @mversic

### PR DESCRIPTION
**Pull request by Arjentix**
## Description

- Fixed #[schema(transparent)]` for structs
- Added transparent type inference

### Linked issue

- Closes #3548  

### Checklist

- [x] I've read `CONTRIBUTING.md`
- [x] I've used the standard signed-off commit format (or will squash just before merging)
- [x] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

**Comment by appetrosyan** [comment]
> Added transparent type inference for
?

**Review by mversic — APPROVED** [review]

File: schema/derive/src/lib.rs [hunk 1]
```diff
@@ -150,6 +189,78 @@ fn impl_into_schema(input: &DeriveInput) -> TokenStream2 {
...
 
+fn infer_transparent_type(input: &Data) -> &syn::Type {
+    const TRY_MESSAGE: &str =
+        "try to specify it explicitly using #[schema(transparent = \"Type\")]";
+
+    match input {
+        Data::Enum(data_enum) => {
+            assert!(
+                data_enum.variants.len() == 1,
+                "Enums with only one variant support transparent type inferring, {}",
```
Comment (appetrosyan) [inline schema/derive/src/lib.rs:200]:
```suggestion
                "Enums with only one variant support transparent type inference, {}",
```

File: schema/derive/src/lib.rs [hunk 2]
```diff
@@ -150,6 +189,78 @@ fn impl_into_schema(input: &DeriveInput) -> TokenStream2 {
...
+                TRY_MESSAGE
+            );
+
+            let variant = data_enum.variants.iter().next().unwrap();
+            let syn::Fields::Unnamed(unnamed_fields) = &variant.fields else {
+                panic!(
+                    "Only unnamed fields are supported for transparent type inferring, {}",
```
Comment (appetrosyan) [inline schema/derive/src/lib.rs:207]:
```suggestion
                    "Only unnamed fields are supported for transparent type inference, {}",
```

File: schema/derive/src/lib.rs [hunk 3]
```diff
@@ -150,6 +189,78 @@ fn impl_into_schema(input: &DeriveInput) -> TokenStream2 {
...
+                    TRY_MESSAGE,
+                )
+            };
+
+            assert!(
+                unnamed_fields.unnamed.len() == 1,
+                "Enums with only one unnamed field support transparent type inferring, {}",
```
Comment (appetrosyan) [inline schema/derive/src/lib.rs:214]:
```suggestion
                "Enums with only one unnamed field support transparent type inference, {}",
```

File: schema/derive/src/lib.rs [hunk 4]
```diff
@@ -150,6 +189,78 @@ fn impl_into_schema(input: &DeriveInput) -> TokenStream2 {
...
+
+            &field.ty
+        }
+        Data::Struct(DataStruct {
+            fields: Fields::Named(fields),
+            ..
+        }) => {
+            assert!(
+                fields.named.len() == 1,
+                "Structs with only one named field support transparent type inferring, {}",
```
Comment (appetrosyan) [inline schema/derive/src/lib.rs:227]:
```suggestion
                "Structs with only one named field support transparent type inference, {}",
```

File: schema/derive/src/lib.rs [hunk 5]
```diff
@@ -150,6 +189,78 @@ fn impl_into_schema(input: &DeriveInput) -> TokenStream2 {
...
+            let field = fields.named.iter().next().unwrap();
+            &field.ty
+        }
+        Data::Struct(DataStruct {
+            fields: Fields::Unnamed(unnamed_fields),
+            ..
+        }) => {
+            assert!(
+                unnamed_fields.unnamed.len() == 1,
+                "Structs with only one unnamed field support transparent type inferring, {}",
```
Comment (appetrosyan) [inline schema/derive/src/lib.rs:239]:
```suggestion
                "Structs with only one unnamed field support transparent type inference, {}",
```

File: schema/derive/src/lib.rs [hunk 6]
```diff
@@ -150,6 +189,78 @@ fn impl_into_schema(input: &DeriveInput) -> TokenStream2 {
...
+            let field = unnamed_fields.unnamed.iter().next().unwrap();
+
+            &field.ty
+        }
+        Data::Struct(DataStruct {
+            fields: Fields::Unit,
+            ..
+        }) => {
+            panic!(
+                "Transparent attribute type inferring is not supported for unit structs, {}",
```
Comment (appetrosyan) [inline schema/derive/src/lib.rs:251]:
```suggestion
                "Transparent attribute type inference is not supported for unit structs, {}",
```

File: schema/derive/src/lib.rs [hunk 7]
```diff
@@ -150,6 +189,78 @@ fn impl_into_schema(input: &DeriveInput) -> TokenStream2 {
...
+                TRY_MESSAGE,
+            )
+        }
+        Data::Union(_) => {
+            panic!(
+                "Transparent attribute type inferring is not supported for unions, {}",
```
Comment (appetrosyan) [inline schema/derive/src/lib.rs:257]:
```suggestion
                "Transparent attribute type inference is not supported for unions, {}",
```

File: schema/derive/src/lib.rs [hunk 8]
```diff
@@ -150,6 +189,78 @@ fn impl_into_schema(input: &DeriveInput) -> TokenStream2 {
...
+        Data::Struct(DataStruct {
+            fields: Fields::Named(fields),
+            ..
+        }) => {
+            assert!(
+                fields.named.len() == 1,
+                "Structs with only one named field support transparent type inferring, {}",
+                TRY_MESSAGE
+            );
+            let field = fields.named.iter().next().unwrap();
```
Comment (appetrosyan) [inline schema/derive/src/lib.rs:230]:
```suggestion
            let field = fields.named.iter().next().expect("Checked via `len`");
```

File: schema/derive/src/lib.rs [hunk 9]
```diff
@@ -150,6 +189,78 @@ fn impl_into_schema(input: &DeriveInput) -> TokenStream2 {
...
+        Data::Struct(DataStruct {
+            fields: Fields::Unnamed(unnamed_fields),
+            ..
+        }) => {
+            assert!(
+                unnamed_fields.unnamed.len() == 1,
+                "Structs with only one unnamed field support transparent type inferring, {}",
+                TRY_MESSAGE
+            );
+            let field = unnamed_fields.unnamed.iter().next().unwrap();
```
Comment (appetrosyan) [inline schema/derive/src/lib.rs:242]:
```suggestion
            let field = unnamed_fields.unnamed.iter().next().expect("Checked via `len`");
```

**Review by mversic — APPROVED** [review]

**Review by mversic — APPROVED** [review]
This is how it should be. Transparent type should not be in the schema. I think we can also infer this from `repr(transparent)` but I wouldn't do that

**Pull request merged**